docs/protocol/specifications/blockchain.md

```diff
@@ -0,0 +1,613 @@
+Program Arguments        | List\<String\>    | List of [signatures](#signature) and other data satisfying previous block’s next consensus program.
+
+#### Block Header Validation
+
+**Inputs:** 
+
+1. BlockHeader entry,
+2. BlockHeader entry from the previous block, `PrevBlockHeader`.
+3. List of transactions included in block.
+
+**Algorithm:**
+
+1. Verify that the block’s version is greater or equal the block version in the previous block header.
+2. Verify that `Height` is equal to `PrevBlockHeader.Height + 1`.
+4. Verify that `PreviousBlockID` is equal to the entry ID of `PrevBlockHeader`.
+5. Verify that `Timestamp` is greater than `PrevBlockHeader.Timestamp`.
+6. Evaluate program `PreviousBlockID.NextConsensusProgram` with [VM version 1](vm1.md) and expansion flag set to `false`. If program execution failed, fail validation.
+7. For each transaction in the block:
+    1. [Validate transaction](#transaction-header-validation) with the timestamp and block version of the input block header.
+8. Compute the [transactions merkle root](#transactions-merkle-root) for the block.
+9. Verify that the computed merkle tree hash is equal to `TransactionsMerkleRoot`.
+10. If the block version is 1: verify that the `ExtHash` is the all-zero hash.
+
+### Block ID
+
+Block ID is defined as an [Entry ID](#entry-id) of the [block header](#block-header) structure.
+
+### Transaction Header
+
+Field      | Type                 | Description
+-----------|----------------------|----------------
+Type       | String               | "txheader"
+Body       | Struct               | See below.  
+Witness    | Struct               | Empty struct.
+
+Body Field | Type                                    | Description
+-----------|-----------------------------------------|-------------------------
+Version    | Integer                                 | Transaction version, equals 1.
+Results    | List\<Pointer\<Output 1\|Retirement 1\>\>   | A list of pointers to Outputs or Retirements. This list must contain at least one item.
+Data       | Hash                                    | Hash of the reference data for the transaction, or a string of 32 zero-bytes (representing no reference data).
+Mintime    | Integer                                 | Must be either zero or a timestamp lower than the timestamp of the block that includes the transaction
+Maxtime    | Integer                                 | Must be either zero or a timestamp higher than the timestamp of the block that includes the transaction.
+ExtHash    | [ExtStruct](#extension-struct)                 | Hash of all extension fields. (See [Extstruct](#extension-struct).) If `Version` is known, this must be 32 zero-bytes.
+
+### Transaction ID
+
+Transaction ID is defined as an [Entry ID](#entry-id) of the [transaction header](#transaction-header) structure.
+
+
+#### Transaction Header Validation
+
+**Inputs:**
+
+1. TxHeader entry,
+2. timestamp,
```

Comment:
...but it would mean that our tx-submission flow, which validates the proposed tx, would have to guess at the next block's timestamp and version, and will occasionally get it wrong, which means it will accept some txs that will eventually be rejected. Which I guess isn't too different from the situation we now have.